src/v2/Components/Artwork/Details.tsx

```diff
@@ -263,11 +265,24 @@ export const LotCloseInfo: React.FC<LotCloseInfoProps> = ({
   saleArtwork,
   sale,
 }) => {
-  const { endAt, extendedBiddingEndAt } = saleArtwork
+  const { endAt, extendedBiddingEndAt, lotID } = saleArtwork
   const biddingEndAt = extendedBiddingEndAt ?? endAt
 
+  const [updatedBiddingEndAt, setUpdatedBiddingEndAt] = useState(biddingEndAt)
+  const [isExtended, setIsExtended] = useState(!!extendedBiddingEndAt)
+
+  const { data } = useWebsocketContext()
+  const { lot_id, extended_bidding_end_at } = data
+  const receivedMessageForThisLot = lot_id === lotID
+  useEffect(() => {
+    if (receivedMessageForThisLot) {
+      setUpdatedBiddingEndAt(extended_bidding_end_at)
+      setIsExtended(true)
+    }
+  }, [receivedMessageForThisLot, extended_bidding_end_at])
```

Comment:
This could be wrapped nicely into a sub-hook that lives in `v2/Apps/Auction/Hooks/useAuctionWebsocket.ts`, something like: 

```ts
const useAuctionWebsocket = ({ onChange, lotID }) => {
  const { data: { lot_id, extended_bidding_end_at } } = useWebsocketContext()
  const receivedMessageForThisLot = lot_id === lotID
  
  useEffect(() => {
    if (receivedMessageForThisLot) {
      onChange(data)
    }
  }, [receivedMessageForThisLot, extended_bidding_end_at])
}
```

And then use the hook like so:

```ts
useAuctionWebsocket({
  lotID,
  onChange: (data) => {
    setUpdatedBiddingAt(data.extended_bidding_end_at)
    setIsExtended(true)
  }
})
```

Reply:
This would also make it real easy to write a test for `useAuctionWebsocket` (which you should do!) since you could just mock dependencies.

Reply:
True this same pattern is used in a few components now!

Reply:
Yea, that wound up working pretty nicely and, while a simple spec, its nice to have and it made it much easier to write.

Reply:
nice!